### PR TITLE
feat: add hide own repository filter checkbox

### PR DESCRIPTION
**Pull request by theflucs**

### CODE REVIEW

File: src/pages/stats/[login].tsx [hunk 1]
```diff
@@ -26,19 +28,34 @@ export default function Stats() {
...
   const filteredRepositories = useMemo(() => {
-    // putting the default value of the repositories
     if (!searchQuery) {
-      return repositories;
+      return hideOwnRepo
+        ? repositories?.filter(
+            // apply hideOwnRepo filter if checked
+            (repoData) =>
+              repoData.repository.owner.login !== session?.user.login
+          )
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:37]: The logic that filters out own repos is written here and replicated in lines 48-51.

Maybe we should either run it once on top of this function or extract the filtering logic in a separate function, what do you think?

File: src/pages/stats/[login].tsx [hunk 2]
```diff
@@ -225,6 +241,16 @@ export default function Stats() {
               onChange={(e) => setSearchQuery(e.target.value)}
             />
           </div>
+          <div className="flex sm:items-start items-center">
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:244]: Here `flex-direction` is `row` (by default), you should probably use the `justify` property.

image

**Review by Balastrong — CHANGES_REQUESTED** [review]

File: src/pages/stats/[login].tsx (hunk 2 above)
Comment (theflucs) [reply src/pages/stats/[login].tsx:244]: are you seeing them centered?
I see them all left-aligned
Screenshot 2024-01-26 at 09 13 04

Comment (Balastrong) [reply src/pages/stats/[login].tsx:244]: They get centered on mobile/smaller screens

File: src/pages/stats/[login].tsx [hunk 3]
```diff
@@ -28,30 +28,38 @@ export default function Stats() {
...
+
+      const filteredReposBySearchQuery = filterReposBySearchQuery(repositories);
+      const filteredOutOwnRepos = filterOutOwnRepos(filteredReposBySearchQuery);
 
-    // filter repositories based on search query
-    const query = searchQuery.toLowerCase();
-    const filterRepos = repositories?.filter((repoData) =>
-      repoData.repository.name.toLowerCase().includes(query)
-    );
+      return hideOwnRepo ? filteredOutOwnRepos : filteredReposBySearchQuery;
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:58]: Here you're running `filterOutOwnRepos` in any case, but if `hideOwnRepo` is false it is unnecessary.

I'd probably see this better with an if statement to return early, or if we want to keep it more functional with the ternary you should run `filterOutOwnRepos` directly inside the "then" of the ternary.

**Review by Balastrong — CHANGES_REQUESTED** [review]

**Review by Balastrong — APPROVED** [review]
LGTM, thank you!

**Pull request merged**